### DISCUSSION TITLE
What if batched update contain breaking dependency

### DISCUSSION THREAD

**Discussion by Paul-Yves** [renovatebot/renovate#13959] [Request Help]
Hello

I am not currently using renovate but dependabot, however I am very interested in renovate for the batch update function.
My question is : if the batched PR contains a dependency that we don't want (breaking change we are not ready for), what is the recommended way to get a new PR with all dependencies execpt this one ?

Should I modify the conf adding the package in the `ignoreDeps` option then rebase the PR or is there another way ?

**Comment by rarkins** [comment]
It's fairly uncommon that a package would have permanently breaking changes in a non-major release. So first of all, don't group major updates unless they are related packages, such as from the same monorepo.

If there's an accidentally breaking changes in a non-major update within a grouped PR, then you could wait for a subsequent patch to be released to fix the PR automatically. Unlike Dependabot, Renovate does not close existing PRs and create replacements and instead updates existing PRs in place, so it's not too "noisy".

Alternatively you could use `ignoreDeps` or perhaps a custom package rule with `matchPackageNames` and `allowedVersions` to disallow just the specific breaking version and not all. You could also alternatively exclude that package from the grouping permanently if they are issuing breaking changes. In general, I don't recommend you group together dependencies if they have a habit of breaking things - keep the group for the reliable ones.
